python/rmm/_lib/memory_resource.pyx

```diff
@@ -0,0 +1,292 @@
+import cython
+import os
+
+from libcpp.memory cimport unique_ptr, make_unique, shared_ptr, make_shared
+from libcpp.string cimport string
+
+
+cdef class MemoryResource:
+    cdef shared_ptr[device_memory_resource] c_obj
+
+
+cdef class CudaMemoryResource(MemoryResource):
+    """
+    Memory resource that uses cudaMalloc/Free for allocation/deallocation
+    """
+    def __cinit__(self):
+        self.c_obj.reset(new cuda_memory_resource())
+
+
+cdef class ManagedMemoryResource(MemoryResource):
+    """
+    Memory resource that uses cudaMallocManaged/Free for
+    allocation/deallocation.
+    """
+    def __cinit__(self):
+        self.c_obj.reset(new managed_memory_resource())
+
+
+cdef class CNMemMemoryResource(MemoryResource):
+    """
+    Memory resource that uses the cnmem pool sub-allocator.
+
+    Parameters
+    ----------
+    initial_pool_size : int, optional
+        Initial pool size in bytes. By default, an implementation defined
+        pool size is used.
+    devices : list of int
+        List of GPU device IDs to register with CNMEM.
+    """
+    def __cinit__(self, size_t initial_pool_size=0, vector[int] devices=[]):
```

Comment:
It's probably safe because of Cython, but normally in Python land I'd say we should make this an empty tuple instead of an empty list to avoid mutation of default arguments.

Maybe check the generated cpp to be sure?

Reply:
Fixed docs to `tuple`